### PR TITLE
LINQ: Fixes ArgumentNullException with StringComparison sensitive case

### LINKED ISSUE #1929: ArgumentNullException with StringComparison.InvariantCulture
**Describe the bug**
Using the [Equals](https://docs.microsoft.com/fr-fr/dotnet/api/system.string.equals?view=netcore-3.1#System_String_Equals_System_String_System_StringComparison_) method in a where clause with `StringComparison.InvariantCulture` an `ArgumentNullException` is throw :

```
System.ArgumentNullException: Value cannot be null. (Parameter 'arguments must not have null items.')
   at Microsoft.Azure.Cosmos.SqlObjects.SqlFunctionCallScalarExpression..ctor(SqlIdentifier name, Boolean isUdf, ImmutableArray`1 arguments)
   at Microsoft.Azure.Cosmos.SqlObjects.SqlFunctionCallScalarExpression.Create(SqlIdentifier name, Boolean isUdf, ImmutableArray`1 arguments)
   at Microsoft.Azure.Cosmos.SqlObjects.SqlFunctionCallScalarExpression.Create(String name, Boolean isUdf, ImmutableArray`1 arguments)
   at Microsoft.Azure.Cosmos.SqlObjects.SqlFunctionCallScalarExpression.Create(String name, Boolean isUdf, SqlScalarExpression[] arguments)
   at Microsoft.Azure.Cosmos.SqlObjects.SqlFunctionCallScalarExpression.CreateBuiltin(String name, SqlScalarExpression[] arguments)
   at Microsoft.Azure.Cosmos.Linq.StringBuiltinFunctions.StringEqualsVisitor.VisitImplicit(MethodCallExpression methodCallExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.BuiltinFunctionVisitor.Visit(MethodCallExpression methodCallExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.StringBuiltinFunctions.Visit(MethodCallExpression methodCallExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.BuiltinFunctionVisitor.VisitBuiltinFunctionCall(MethodCallExpression methodCallExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitNonSubqueryScalarExpression(Expression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitNonSubqueryScalarExpression(Expression expression, ReadOnlyCollection`1 parameters, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitScalarExpression(Expression expression, ReadOnlyCollection`1 parameters, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitScalarExpression(LambdaExpression lambda, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitWhere(ReadOnlyCollection`1 arguments, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.VisitMethodCall(MethodCallExpression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.Translate(Expression inputExpression, TranslationContext context)
   at Microsoft.Azure.Cosmos.Linq.ExpressionToSql.TranslateQuery(Expression inputExpression, IDictionary`2 parameters, CosmosSerializationOptions serializationOptions)
   at Microsoft.Azure.Cosmos.Linq.SqlTranslator.TranslateQuery(Expression inputExpression, CosmosSerializationOptions serializationOptions, IDictionary`2 parameters)
   at Microsoft.Azure.Cosmos.Linq.DocumentQueryEvaluator.HandleMethodCallExpression(MethodCallExpression expression, IDictionary`2 parameters, CosmosSerializationOptions serializationOptions)
   at Microsoft.Azure.Cosmos.Linq.DocumentQueryEvaluator.Evaluate(Expression expression, CosmosSerializationOptions serializationOptions, IDictionary`2 parameters)
   at Microsoft.Azure.Cosmos.Linq.CosmosLinqQuery`1.CreateFeedIterator(Boolean isContinuationExcpected)
   at Microsoft.Azure.Cosmos.Linq.CosmosLinqQuery`1.ToFeedIterator()
   at Microsoft.Azure.Cosmos.Linq.CosmosLinqExtensions.ToFeedIterator[T](IQueryable`1 query)
```

There is no exception with `StringComparison.InvariantCultureIgnoreCase`

**To reproduce**
```cs
var cosmosClient = new CosmosClient(connectionString);
var database = cosmosClient.GetDatabase("db1");
var container = database.GetContainer("container1");
var queryable = container.GetItemLinqQueryable<ToDoItem>();
var iterator = queryable.Where(x => x.Name.Equals(subjectId, StringComparison.InvariantCulture)).ToFeedIterator();
```

**Environment summary**
SDK Version: 3.14.0
OS Version: Windows 10 1909

### PR DESCRIPTION
**Pull request by ylabade**
# Pull Request Template

## Description

This PR addresses the issue #1929

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Closing issues

To automatically close an issue: closes #1929

### CODE REVIEW

**Comment by ealsur** [comment]
@khdang Can you please review this LINQ fix?

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs [hunk 1]
```diff
@@ -166,21 +166,28 @@ private sealed class SqlStringWithComparisonVisitor : BuiltinFunctionVisitor
                 StringComparison.OrdinalIgnoreCase
             });
 
+            private static readonly HashSet<StringComparison> SensitiveCaseComparisons = new HashSet<StringComparison>(new[]
+            {
+                StringComparison.CurrentCulture,
+                StringComparison.InvariantCulture,
+                StringComparison.Ordinal
+            });
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:174]: Can you convert these to immutable hash sets now that we support the types?

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs [hunk 2]
```diff
@@ -166,21 +166,28 @@ private sealed class SqlStringWithComparisonVisitor : BuiltinFunctionVisitor
...
+            public static SqlScalarExpression GetCaseSensitivityExpression(Expression expression)
             {
                 if (expression is ConstantExpression inputExpression
-                    && inputExpression.Value is StringComparison comparisonValue
-                    && IgnoreCaseComparisons.Contains(comparisonValue))
+                    && inputExpression.Value is StringComparison comparisonValue)
                 {
-                    SqlBooleanLiteral literal = SqlBooleanLiteral.Create(true);
-                    return SqlLiteralScalarExpression.Create(literal);
+                    return SensitiveCaseComparisons.Contains(comparisonValue) ? SqlLiteralScalarExpression.Create(SqlBooleanLiteral.Create(false))
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs:188]: nit: an if else ladder might be better here as opposed to nested ternary operators.

File: Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestSpecialMethods.xml [hunk 3]
```diff
@@ -10,6 +10,19 @@ SELECT VALUE (root["StringField"] = "str")
 FROM root]]></SqlQuery>
     </Output>
   </Result>
+  <Result>
+    <Input>
+      <Description><![CDATA[Equals (case-sensitive)]]></Description>
```
Comment (bchong95) [inline Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/BaselineTest/TestBaseline/LinqTranslationBaselineTests.TestSpecialMethods.xml:15]: Thank you for adding these baselines :)

**Review by bchong95 — APPROVED** [review]

File: Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs (hunk 2 above)
Comment (ylabade) [reply Microsoft.Azure.Cosmos/src/Linq/BuiltinFunctions/StringBuiltinFunctions.cs]: done

**Review by bchong95 — APPROVED** [review]

**Pull request merged**